java/query/src/main/java/sleeper/query/lambda/SqsQueryProcessor.java

```diff
@@ -120,12 +119,20 @@ public void processQuery(QueryOrLeafPartitionQuery query) {
     }
 
     private CloseableIterator<Record> processRangeQuery(Query query, TableProperties tableProperties, QueryStatusReportListeners queryTrackers) throws StateStoreException, QueryException {
+        // If the cache needs refreshing remove to allow for a new in initialisation
+        LOGGER.debug("Cache for table {}: {}", query.getTableName(), queryExecutorCache);
+        if (!queryExecutorCache.isEmpty() && queryExecutorCache.get(query.getTableName()).cacheRefreshRequired()) {
+            LOGGER.info("Refreshing Query Executor cache for table {}", query.getTableName());
+            queryExecutorCache.remove(query.getTableName());
+        }
```

Comment:
The caching behaviour here is untested. One way to cover this would be to add a method like `QueryExecutor.initIfNeeded(Instant now)` and test against QueryExecutor (see my comment on QueryExecutor).